lib/xlsxtream/workbook.rb

```diff
@@ -62,9 +62,7 @@ def add_worksheet(*args, &block)
       if block_given?
         # This method used to be an alias for `write_worksheet`. This was never publicly documented,
         # but to avoid breaking this private API we keep the old behaviour when called with a block.
-        if defined?(Warning) # Warning doesn't exist in ruby < 2.4
-          Warning.warn "#{caller.first}: Use #write_worksheet to pass a block, passing a block to add_worksheet is deprecated."
-        end
+        Kernel.warn "#{caller.first[/.*:\d+:(?=in `)/]} warning: Calling #{self.class}#add_worksheet with a block is deprecated, use #write_worksheet instead."
```

Comment:
Ah, good idea, this was better than what I did!

Reply:
Yeah, according to the docs `Warning.warn` is provided to be able to override the default way to log warnings in Ruby, but it doesn't have any logic to check the warning level, so you cannot suppress messages output directly using `Warning.warn`.